src/cgroups.c

```diff
@@ -145,21 +222,58 @@ static int read_cpuacct_root(const char *dirname, const char *filename,
   }
 
   if (S_ISDIR(statbuf.st_mode)) {
-    status = walk_directory(abs_path, read_cpuacct_procs,
+    status = walk_directory(abs_path, callback,
                             /* user_data = */ NULL,
                             /* include_hidden = */ 0);
     return status;
   }
 
   return 0;
-}
+} /* int read_cgroups_root */
+
+/*
+ * Gets called for every file/folder in /sys/fs/cgroup/cpu,cpuacct (or
+ * wherever cpuacct is mounted on the system). Calls walk_directory with the
+ * read_cpuacct_procs callback on every folder it finds, such as "system".
+ */
+__attribute__((nonnull(1, 2))) static int
+read_cpuacct_root(const char *dirname, const char *filename, void *user_data) {
+  return read_cgroups_root(dirname, filename, read_cpuacct_procs, user_data);
+} /* int read_cpuacct_root */
+
+/*
+ * Gets called for every file/folder in /sys/fs/cgroup/memory (or
+ * wherever memory is mounted on the system). Calls walk_directory with the
+ * read_memory_procs callback on every folder it finds, such as "system".
+ */
+__attribute__((nonnull(1, 2))) static int
+read_memory_root(const char *dirname, const char *filename, void *user_data) {
+  return read_cgroups_root(dirname, filename, read_memory_procs, user_data);
+} /* int read_memory_root */
+
+/*
+ * Gets called for every file/folder in /sys/fs/cgroup (or
+ * wherever cgroup2 is mounted on the system). Calls walk_directory with the
+ * read_cpu_procs callback on every folder it finds, such as "system".
+ */
+__attribute__((nonnull(1, 2))) static int
+read_cpu_root(const char *dirname, const char *filename, void *user_data) {
+  return read_cgroups_root(dirname, filename, read_cpu_procs, user_data);
+} /* int read_cpu_root */
 
 static int cgroups_init(void) {
   if (il_cgroup == NULL)
     il_cgroup = ignorelist_create(1);
 
   return 0;
-}
+  return (0);
```

Comment:
left-over return (0) here?

In general, we use return 0 instead of (0). Could you please adjust your changes?